code/modules/heavy_vehicle/mech_life.dm

```diff
@@ -41,6 +41,20 @@
 	handle_vision()
 	handle_hud_icons()
 
+/mob/living/heavy_vehicle/think()
+	if(length(pilots))
+		return
```

Comment:
If there is a pilot the following status must be cleared so the mech doesn't dart off to somewhere in case the pilot gets out.

Reply:
i made it so that following is kept if the person is the pilot, so you can hop in and out without having to reset your follow status